scrapinghub/client/jobs.py

```diff
@@ -55,96 +67,168 @@ def count(self, **params):
+            - get certain number of last finished jobs per some spider::
 
-            >>> jobs_summary = project.jobs.iter(
-            ...     spider='spider2', state='finished', count=3)
+                >>> jobs_summary = project.jobs.iter(
+                ...     spider='spider2', state='finished', count=3)
         """
+        filter_kwargs = [('spider', spider), ('state', state),
+                         ('count', count), ('start', start),
+                         ('has_tag', has_tag), ('lacks_tag', lacks_tag),
+                         ('startts', startts), ('endts', endts),
+                         ('jobmeta', meta)]
+        params.update({k: v for k, v in filter_kwargs if v is not None})
         if self.spider:
             params['spider'] = self.spider.name
         return self._project.jobq.list(**params)
 
-    def list(self, **params):
+    def list(self, count=None, start=None, spider=None, state=None,
+             has_tag=None, lacks_tag=None, startts=None, endts=None,
+             meta=None, **params):
         """Convenient shortcut to list iter results.
 
+        :param count: (optional) limit amount of returned jobs.
+        :param start: (optional) number of jobs to skip in the beginning.
+        :param spider: (optional) filter by spider name.
+        :param state: (optional) a job state, a string or a list of strings.
+        :param has_tag: (optional) filter results by existing tag(s), a string
+            or a list of strings.
+        :param lacks_tag: (optional) filter results by missing tag(s), a string
+            or a list of strings.
+        :param startts: (optional) UNIX timestamp at which to begin results,
+            in millisecons.
+        :param endts: (optional) UNIX timestamp at which to end results,
+            in millisecons.
+        :param meta: (optional) request for additional fields, a single
+            field name or a list of field names to return.
+        :param \*\*params: (optional) other filter params.
+
+        :return: list of dictionaries of jobs summary for a given filter params
+        :rtype: list[dict].
+
         Please note that list() method can use a lot of memory and for a large
         amount of jobs it's recommended to iterate through it via iter()
         method (all params and available filters are same for both methods).
-
         """
+        # FIXME we double-check the params here, is there a better way?
+        # Simpler way would be to keep **params only here and point to iter(),
+        # but then we loose hinting kwargs for list() method.
+        filter_kwargs = [('spider', spider), ('state', state),
+                         ('count', count), ('start', start),
+                         ('has_tag', has_tag), ('lacks_tag', lacks_tag),
+                         ('startts', startts), ('endts', endts),
+                         ('meta', meta)]
+        params.update({k: v for k, v in filter_kwargs if v is not None})
```

Comment:
possibly you can generalize filtering? something like

```
update_kwargs(kwargs, spider=spider, state=state, count=count,
               spidername=spider_name)
```

note the last one, you can use map to different name if necessary